docs/user-guide/steerlm.rst

```diff
@@ -50,6 +50,8 @@ This section is a step-by-step tutorial that walks you through how to run a full
 
 4. Inference on the SteerLM model with different attribute values
 
```

Comment:
Line 9/9 fix capitalization and add acronym for terms. 

**SteerLM** is a novel approach developed by the NVIDIA NeMo Team, introduced as part of NVIDIA NeMo Alignment methods. It simplifies the customization of large language models (LLMs) and empowers users with dynamic control over model outputs by specifying desired attributes. 

Despite remarkable progress in natural language generation driven by LLMs like GPT-3, Megatron-Turing, Chinchilla, PaLM-2, Falcon, and Llama 2, these foundational models often fall short in delivering nuanced and user-aligned responses. 

The current approach for LLM improvement combines Supervised Fine-Tuning (SFT) and Reinforcement Learning from Human Feedback (RLHF), but it comes with complexities and limited user control. 

SteerLM addresses these challenges and represents a significant advancement in the field, making it easier to tailor LLMs to specific needs and preferences. This document delves into how SteerLM operates and offers guidance on training a SteerLM model.

Line 13/13 use SFT acronym previously defined.

SteerLM leverages a SFT method that empowers you to control responses during inference. It overcomes the limitations of prior alignment techniques, and consists of four key steps:

Line 32 use RLHF acronym previously defined.

RLHF and SteerLM are two methods aimed at aligning language models to human preferences. RLHF trains language models by providing positive or negative feedback on generated responses, reinforcing good behaviors. Specifically, the model is encouraged to generate more text similar to responses that receive positive feedback, and less like those with negative feedback.

Line 34 revise.

The two methods approach model alignment from different angles: RLHF reinforces desired model behaviors directly, while SteerLM steers generation based on categorical labels. Both aim to produce language model outputs that are better aligned with human values and preferences.

Line 36 revise preposition.

.. note::
   For details on SteerLM, please refer to our paper `SteerLM: Attribute Conditioned SFT as an (User-Steerable) Alternative to RLHF <https://arxiv.org/abs/2310.05344>`_.

Line 38 revise.

For details about the HelpSteer dataset, please refer to our paper `HelpSteer: Multi-attribute Helpfulness Dataset for SteerLM <https://arxiv.org/abs/2311.09528>`_.

Line 43/43 revise.

This section is a step-by-step tutorial that walks you through how to run a full SteerLM pipeline with a Llama2 70B LLM model. 

Line 51/51 Delete this list of 4 steps. It is redundant and not needed. Also, the list contains only 4 steps and the procedures that follow contain 6 steps. It is best to be consistent here. Not needed, so remove the steps here.

Line 54/56 - 55/57 remove step (this is a procedural heading that contains numbered steps), fix capitalization, and change to sub-sub heading level.


Download the Llama 2 LLM Model
	
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 56/58 change to a numbered step in this section. (Ok to use numbers like 1., 2., etc. or #. character for steps. I use numbers below as an example).

1. Download the Llama 2 70B LLM model from HF <https://huggingface.co/meta-llama/Llama-2-70b-hf> into the models folder.

Line 58/60 revise and change to a numbered step.

2. Convert the Llama 2 LLM into .nemo format:


Line 65/67 change to a numbered step, revise, and fix spelling.


3. Download and convert to .nemo format for the 13B model <https://huggingface.co/meta-llama/Llama-2-13b-hf>. This is needed for the Attribute Prediction Modeling step.

67/69 change to a numbered step.

4. Untar the .nemo file to obtain the tokenizer in NeMo format (only for the 70B model):

	
Line 79/81 - 80/82 remove step, fix capitalization, and change to sub-sub heading level.


Download and Preprocess Data for Attribute Prediction Modeling 
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 82/84 revise, fix punctuation, and change to a numbered step in this section.


1. Download and convert both datasets into a common format:

Line 90/92 revise, fix punctuation, and change to a numbered step.

2. Merge the two datasets for the train and val subset, respectively:

Line 98/100 revise, fix punctuation, and change to a numbered step.

3. Preprocess the data into regression reward model training format:


Line 111/113 - 112/114 remove step, fix capitalization, and change to sub-sub heading level.


Train the Regression Reward Model on OASST+HelpSteer Data
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Line 116/118 change to note format, revise, and fix punctuation.

.. note::
  Depending on the type of cluster you use, you may need to set up multi-node training in your cluster env. For details, please refer to https://lightning.ai/docs/pytorch/stable/clouds/cluster.html. 


Line 146/148 - 147/149 remove step, fix capitalization, and change to sub-sub heading level.


Generate Annotations
^^^^^^^^^^^^^^^^^^^^^
 
Line 148/150 change to a numbered step in this section.

1. To generate annotations, run the following command in the background to launch an inference server:

Line 162/164 change to a numbered step.

2. Execute the following code:

Line 179/181 - 180/182 remove step, fix capitalization, and change to sub-sub heading level.


Train the Attribute-Conditioned SFT Model
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

236/238 - 237/239 remove step, revise title (include a verb) and change to sub-sub heading level.

Run Inference
^^^^^^^^^^^^^^

Line 238/240 change to numbered step in this section.

1. To start inference, run an inference server in the background using the following command:

Line 256/258 change to a numbered step.

2. Create Python helper functions:

Line 288/290 change to a numbered step.

3. Change the values below to steer the language model:

Line 307/309 change to a numbered step.

4. Ask questions and generate responses: